Tests/conftest.py

```diff
@@ -10,3 +10,18 @@ def pytest_report_header(config):
             return out.getvalue()
     except Exception as e:
         return f"pytest_report_header failed: {e}"
+
+def pytest_configure(config):
+    # We're marking some tests to ignore valgrind errors and XFAIL them.
+    # Ensure that the mark is defined even in cases where pytest-valgrind isn't installed
+
+    import pytest
+    import warnings
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        try:
+            getattr(pytest.mark, "valgrind_known_error")
+        except:
```

Comment:
?
```suggestion
        except AttributeError:
```

Reply:
I don't think it's an attribute error, as it's catching a warning thrown by pytest which hooks the getattr to specifically avoid an attribute error.